WiseUnpacker/SabreTools.Serialization.Deserializers/WiseSectionHeader.cs

```diff
@@ -87,41 +92,210 @@ public class WiseSectionHeader : BaseBinaryDeserializer<SectionHeader>
+            
+            if (headerLength > 17)
+            {
+                header.UnknownValue18 = data.ReadUInt32LittleEndian();
+            }
+            
+            // Parse strings
+            // TODO: Count size of string section for later size verification
 
-            // TODO: Parse strings
-            header.Strings = null;
+            PreStringValuesHelper(data, header, initialOffset, localWisOffset, header.Version, out int preStringBytesSize);
+            
+            byte[][]? stringArrays = StringHelper(data, header, initialOffset, localWisOffset, preStringBytesSize);
+            if (stringArrays == null)
+            {
+                return null;
+            }
+            header.Strings = stringArrays;
+            
+            // Should really be done in the wrapper, but almost everything there is static so there's no good place
+            if (header.UnknownDataSize != 0) // Not sure what this data is. Might be a wisescript?
+            {
+                data.Seek(data.Position + header.UnknownDataSize, 0);
+            }
 
             return header;
         }
+
+        private static bool PreStringValuesHelper(Stream data, SectionHeader header, long initialOffset, int localWisOffset, byte[] version, out int preStringBytesSize)
+        {
+            data.Seek(initialOffset + localWisOffset, 0);
+            header.TmpString = data.ReadNullTerminatedAnsiString();
+            header.GuidString = data.ReadNullTerminatedAnsiString();
+            // TODO: better way to figure out how far it's needed to advance?
+            int versionSize;
+            if (version[version.Length - 1] == 0x02)
+                versionSize = version[version.Length - 3];
+            else
+                versionSize = version[version.Length - 2];
+            
+            if (versionSize <= 1) // third byte seems to indicate size of NonWiseVer
+            {
+                byte[] stringBytes = data.ReadBytes(versionSize);
+                header.NonWiseVersion = Encoding.ASCII.GetString(stringBytes);
+                if (localWisOffset <= 77)
+                {
+                    header.PreFontValue = data.ReadBytes(2);
+                }
+                else
+                {
+                    header.PreFontValue = data.ReadBytes(4);
+                }
+            }
+            
+            else // If that third byte is 0x01, no NonWiseVersion string is present.
+            {
+                header.PreFontValue = data.ReadBytes(3);
+            }
+            
+            header.FontSize = data.ReadByte(); 
+            preStringBytesSize = WiseSectionPreStringBytesSize[localWisOffset];
+            if (version[1] == 0x01)
+            {
+                preStringBytesSize = 2; // hack for Codesited5.exe , very early and very strange.
+            }
+            
+            return true;
+        }
+
+        private static byte[][]? StringHelper(Stream data, SectionHeader header, long initialOffset,
```

Comment:
If you're going to wrap method parameters, please do them one per line. See other code in the library for examples.

Reply:
Would it be ok to just do them in one line instead, if those are my options?

Reply:
Never mind; two of the variables were unused and I forgot to remove them.

Reply:
Should be fixed.